packages/back-end/src/services/features.ts

```diff
@@ -58,7 +81,14 @@ export async function getFeatureDefinitions(
             const rule: FeatureDefinitionRule = {};
             if (r.condition && r.condition !== "{}") {
               try {
-                rule.condition = JSON.parse(r.condition);
+                rule.condition = JSON.parse(
+                  replaceSavedGroupsInCondition(
+                    r.condition,
+                    groupMap,
+                    allGroups,
```

Comment:
I would simplify this logic a lot.  Instead of passing in `groupMap`, `allGroups` and `attributeMap`, all you really need to do is pass in `groupMap` if you do this:

```ts
type GroupMap = Map<string, string[] | number[]>;

function getGroupValues(values: string[], type?: string): string[] | number[] {
  if (type === "number") {
    return values.map(v => parseFloat(v));
  }
  return values;
}

const groupMap: GroupMap = new Map(
  allGroups.map((group) => {
    const attributeType = attributeMap?.get(group.attributeKey);
    const values = getGroupValues(group.values, attributeType);
    return [group.id, values];
  })
)
```

If you do this, the `replaceSavedGroupsInCondition` function can just do `JSON.stringify` and it will work.

Another benefit of this is it will make the code much easier to test since there will be fewer dependencies.